seacatauth/batman/elasticsearch.py

```
@@ -124,6 +127,20 @@ async def _elasticsearch_session(self):
+	@contextlib.asynccontextmanager
+	async def _with_elasticsearch_nodes(self, api_call: typing.Callable[..., typing.Awaitable]):
+		for node_url in random.sample(self.ElasticSearchNodesUrls, len(self.ElasticSearchNodesUrls)):
+			async with self._elasticsearch_session(base_url=node_url) as session:
+				try:
+					result = await api_call(session=session)
+					yield result
+				except (aiohttp.client_exceptions.ClientConnectionError, asyncio.TimeoutError):
+					L.debug("ElasticSearch node {} is not reachable, trying another one.".format(node_url))
+					continue
+		raise aiohttp.client_exceptions.ClientConnectionError(
+			"Cannot connect to any of the configured ElasticSearch nodes.")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)